builder/amazon/common/regions.go

```diff
@@ -15,23 +15,31 @@ func getValidationSession() *ec2.EC2 {
 	return ec2conn
 }
 
-func listEC2Regions(ec2conn ec2iface.EC2API) []string {
+func listEC2Regions(ec2conn ec2iface.EC2API) ([]string, error) {
 	var regions []string
-	resultRegions, _ := ec2conn.DescribeRegions(nil)
+	resultRegions, err := ec2conn.DescribeRegions(nil)
+	if err != nil {
+		return []string{}, err
+	}
 	for _, region := range resultRegions.Regions {
 		regions = append(regions, *region.RegionName)
 	}
 
-	return regions
+	return regions, nil
 }
 
 // ValidateRegion returns true if the supplied region is a valid AWS
 // region and false if it's not.
-func ValidateRegion(region string, ec2conn ec2iface.EC2API) bool {
-	for _, valid := range listEC2Regions(ec2conn) {
+func ValidateRegion(region string, ec2conn ec2iface.EC2API) (bool, error) {
```

Comment:
Here, I believe the bool is not necessary anymore. When the region is invalid, then an error saying why should be returned.

[PR 6787](https://github.com/hashicorp/packer/pull/6787/files#diff-743b70f36793a4f1edd68908b6048ef1) had something like this.

May be we could cherry pick those commit so we don't have to wait.